theano/sandbox/gpuarray/basic_ops.py

```diff
@@ -312,254 +309,29 @@ def infer_shape(self, node, xshp):
 
     def c_code(self, node, name, inputs, outputs, sub):
         return """
+        PyGpuArrayObject *%(name)s_tmp;
+        %(name)s_tmp = PyArray_GETCONTIGUOUS(%(inp)s);
```

Comment:
I mean by not trying to make it contiguous beforehand.  It finds the lowest and higest elements and copies all of the data in between (even if that data is not part of the array according to strides and dims).

I want to have libgpuarray make a contiguous version, but this PR can't do that since it's a separate repo, hence the issue.